nat-lab/tests/test_mesh_exit_through_peer.py

```diff
@@ -111,3 +143,114 @@ async def test_mesh_exit_through_peer(
+            marks=pytest.mark.windows,
+        ),
+        pytest.param(
+            ConnectionTag.MAC_VM, telio.AdapterType.Default, marks=pytest.mark.mac
+        ),
+    ],
+)
+@pytest.mark.asyncio
+async def test_ipv6_exit_node(
+    alpha_connection_tag: ConnectionTag,
+    adapter_type: AdapterType,
+    exit_ip_stack: IPStack,
+) -> None:
+    async with AsyncExitStack() as exit_stack:
+        api = API()
+
+        (alpha, beta) = api.default_config_two_nodes(
+            alpha_ip_stack=IPStack.IPv4v6, beta_ip_stack=exit_ip_stack
+        )
+        (connection_alpha, alpha_conn_tracker) = await exit_stack.enter_async_context(
+            new_connection_with_conn_tracker(
+                alpha_connection_tag,
+                generate_connection_tracker_config(
+                    alpha_connection_tag, derp_1_limits=ConnectionLimits(1, 1)
+                ),
+            )
+        )
+        (connection_beta, beta_conn_tracker) = await exit_stack.enter_async_context(
+            new_connection_with_conn_tracker(
+                ConnectionTag.DOCKER_OPEN_INTERNET_CLIENT_DUAL_STACK,
+                generate_connection_tracker_config(
+                    ConnectionTag.DOCKER_OPEN_INTERNET_CLIENT_DUAL_STACK,
+                    derp_1_limits=ConnectionLimits(1, 1),
+                    stun_limits=ConnectionLimits(1, 2),
+                ),
+            )
+        )
+
+        client_alpha = await exit_stack.enter_async_context(
+            telio.Client(connection_alpha, alpha, adapter_type).run_meshnet(
+                api.get_meshmap(alpha.id)
+            )
+        )
+
+        client_beta = await exit_stack.enter_async_context(
+            telio.Client(connection_beta, beta).run_meshnet(api.get_meshmap(beta.id))
+        )
+
+        await testing.wait_lengthy(
+            asyncio.gather(
+                client_alpha.wait_for_state_on_any_derp([State.Connected]),
+                client_beta.wait_for_state_on_any_derp([State.Connected]),
+                alpha_conn_tracker.wait_for_event("derp_1"),
+                beta_conn_tracker.wait_for_event("derp_1"),
+            )
+        )
+        await testing.wait_lengthy(
+            asyncio.gather(
+                client_alpha.wait_for_state_peer(beta.public_key, [State.Connected]),
+                client_beta.wait_for_state_peer(alpha.public_key, [State.Connected]),
+            )
+        )
+
+        # Ping in-tunnel node with IPv6
+        async with Ping(
+            connection_alpha,
+            testing.unpack_optional(beta.get_ip_address(IPProto.IPv6)),
+        ).run() as ping6:
+            await testing.wait_long(ping6.wait_for_next_ping())
+
+        await testing.wait_long(client_beta.get_router().create_exit_node_route())
+        await testing.wait_long(client_alpha.connect_to_exit_node(beta.public_key))
+        await testing.wait_long(
+            client_alpha.wait_for_state_peer(beta.public_key, [State.Connected])
+        )
+
```

Comment:
Do we have any ability to do IPv6 STUN? It would be really nice to double-check, that we are getting IPv6 STUNed address when routing through exit node.

Reply:
We can, but we need to change the stun server than, because `stund` does not support ipv6.

Reply:
Create separate task for that (LLT-4310).